crates/hir-def/src/db.rs

```diff
@@ -141,8 +142,16 @@ pub trait DefDatabase:
         id: VariantId,
     ) -> (Arc<VariantFields>, Arc<ExpressionStoreSourceMap>);
 
+    #[salsa::tracked]
+    fn enum_variants(&self, id: EnumId) -> Arc<EnumVariants> {
+        self.enum_variants_with_diagnostics(id).0
+    }
```

Comment:
I assume you changed this in the macro, but what does this do?

(Should we just define this this as a tracked method on `EnumId` instead?)

Reply:
It's just shorthand syntax. Instead of having to create a a function and pointing to it wtih `invoke_actual` this allows writing that inline. Its basically just `salsa::tracked` but written as part of the trait here

Reply:
Ah, gotcha. We can probably replace a lot of usages of `invoke_actual` with this, I think!